climada_petals/engine/supplychain.py

```diff
@@ -19,226 +19,479 @@
+    elif mriot_type == "WIOD16":
+        mriot_df = pd.read_excel(downloaded_file, engine="pyxlsb")
 
-        """
+        Z, Y, x = parse_mriot_from_df(
+            mriot_df,
+            col_iso3=2,
+            col_sectors=1,
+            rows_data=(5, 2469),
+            cols_data=(4, 2468),
+        )
 
-        file_name = 'WIOT{}_Nov16_ROW.xlsb'.format(year)
-        file_loc = WIOD_DIRECTORY / file_name
-        if not file_loc.exists():
-            if not WIOD_DIRECTORY.exists():
-                WIOD_DIRECTORY.mkdir()
-            LOGGER.info('Downloading folder with WIOD tables')
-
-            downloaded_file_name = u_fh.download_file(WIOD_FILE_LINK,
-                                                      download_dir=WIOD_DIRECTORY)
-            downloaded_file_zip_path = Path(downloaded_file_name + '.zip')
-            Path(downloaded_file_name).rename(downloaded_file_zip_path)
-
-            with zipfile.ZipFile(downloaded_file_zip_path, 'r') as zip_ref:
-                zip_ref.extractall(WIOD_DIRECTORY)
-
-        mriot = pd.read_excel(file_loc, engine='pyxlsb')
-
-        start_row, end_row = range_rows
-        start_col, end_col = range_cols
-
-        self.sectors = mriot.iloc[start_row:end_row, col_sectors].unique()
-        self.mriot_reg_names = mriot.iloc[start_row:end_row, col_iso3].unique()
-        self.mriot_data = mriot.iloc[start_row:end_row,
-                                     start_col:end_col].values
-        self.total_prod = mriot.iloc[start_row:end_row, -1].values
-        self.reg_pos = {
-            name: range(len(self.sectors)*i, len(self.sectors)*(i+1))
-            for i, name in enumerate(self.mriot_reg_names)
-            }
-        self.mriot_type = 'WIOD'
-
-    def calc_sector_direct_impact(self, hazard, exposure, imp_fun_set,
-                                  selected_subsec="service"):
-        """Calculate direct impacts.
+        mriot = pymrio.IOSystem(Z=Z, Y=Y, x=x)
+        mriot.unit = "M.EUR"
 
-        Parameters
-        ----------
-        hazard : Hazard
-            Hazard object for impact calculation.
-        exposure : Exposures
-            Exposures object for impact calculation. For WIOD tables, exposure.region_id
-            must be country names following ISO3 codes.
-        imp_fun_set : ImpactFuncSet
-            Set of impact functions.
-        selected_subsec : str or list
-            Positions of the selected sectors. These positions can be either
-            defined by the user by passing a list of values, or by using built-in
-            sectors' aggregations for the WIOD data passing a string with possible
-            values being "service", "manufacturing", "agriculture" or "mining".
-            Default is "service".
+    elif mriot_type == "OECD21":
+        mriot = pymrio.parse_oecd(path=downloaded_file)
+        mriot.x = pymrio.calc_x(mriot.Z, mriot.Y)
 
-        """
+    return mriot
```

Comment:
Variable possibly unbound here
```suggestion
    else:
        raise RuntimeError(f"Unknown mriot_type: {mriot_type}")

    return mriot
```